core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java

```diff
@@ -32,13 +32,15 @@ public AbstractFeeDistribution(Address addressProvider) {
 
 
     protected void distributeFee(BigInteger amount) {
-        BigInteger remaining = amount;
         Address lendingPoolCoreAddr = getAddress(Contracts.LENDING_POOL_CORE.getKey());
         Address daoFundAddr = getAddress(Contracts.DAO_FUND.getKey());
+
+        BigInteger remaining = amount;
+        BigInteger denominator = ICX;
+
         for (Address receiver : feeDistributionWeight.keySet()) {
             BigInteger percentageToDistribute = feeDistributionWeight.get(receiver);
-            BigInteger amountToDistribute = (percentageToDistribute.multiply(amount)).divide(ICX);
-            remaining = remaining.subtract(amountToDistribute);
+            BigInteger amountToDistribute = (percentageToDistribute.multiply(remaining)).divide(denominator);
```

Comment:
Zero denominator can raise "java.lang.ArithmeticException: BigInteger divide by zero".

Reply:
I left my suggestion on the following issue:
https://github.com/ommfinance/openmoneymarket-audit/issues/16#issuecomment-1864943798